homeassistant/components/aqara/base.py

```diff
@@ -0,0 +1,278 @@
+        """Load JSON string and return a IntegerTypeData object."""
+        return cls(**json.loads(data))
+
+
+class DeviceValueRange(SimpleNamespace):
+    """device's value range.
+
+    Attributes:
+        type(str): value's type, which may be Boolean, Integer, Enum, Json
+        values(dict): value range
+    """
+
+    type: str
+    values: str
+
+
+@dataclass
+class EnumTypeData:
+    """Enum Type Data."""
+
+    range: list[str]
+
+    @classmethod
+    def from_json(cls, data: str) -> EnumTypeData:
+        """Load JSON string and return a EnumTypeData object."""
+        return cls(**json.loads(data))
+
+
+class AqaraEntity(Entity):
+    """Aqara base device."""
+
+    _attr_should_poll = False
+
+    def __init__(self, point: AqaraPoint, device_manager: AqaraDeviceManager) -> None:
+        """Init AqaraHaEntity."""
+        self._attr_unique_id = f"Aqara.{point.id}"
+        self.point = point
+        self.device_manager = device_manager
+        device = self.get_aqara_device()
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, string_dot_to_underline(self.point.did))},
+            manufacturer="Aqara",
+            name=f"{device.device_name}({self.point.did})",
+            model=device.model,
+            suggested_area=f"{device.position_name}",
+        )
+
+    def get_aqara_device(self) -> AqaraDevice:
+        """Get the device of the point."""
+        return self.device_manager.get_device(self.point.did)
+
+    @property
+    def name(self) -> str | None:
+        """Return Aqara device name."""
+        if isinstance(self.point.name, str) and self.point.name != "":
+            return f"{self.point.name}({self.point.did})"
+        if hasattr(self, "entity_description"):
+            if self.entity_description.name is not None and isinstance(
+                self.entity_description.name, str
+            ):
+                return f"{self.entity_description.name}({self.point.did})"
+        return f"({self.point.id})"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+        device = self.device_manager.get_device(self.point.did)
+        return DeviceInfo(
+            identifiers={(DOMAIN, string_dot_to_underline(self.point.did))},
+            manufacturer="Aqara",
+            name=f"{device.device_name}({self.point.did})",
+            model=self.device_manager.get_device_model(self.point.did),
+            suggested_area=f"{device.position_name}",
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return if the device is available."""
+        return self.point.is_online()
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity is added to hass."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{AQARA_HA_SIGNAL_UPDATE_ENTITY}_{string_dot_to_underline(self.point.id)}",
```

Comment:
Do we need to use `string_dot_to_underline` here? As in, it's not user-facing, so we can just use `self.point.id` directly here, right?

Reply:
updatd